src/main/java/service/Preprocessor.java

```diff
@@ -0,0 +1,46 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Preprocessor {
+
+    private final Pattern compile = Pattern.compile("\\d+|[+\\-*/]");
```

Comment:
```suggestion
    private static final Pattern compile = Pattern.compile("\\d+|[+\\-*/]");
```
클래스마다 고정된 값을 가져갈테니 클래스 변수로 선언하면 더 좋을 것 같습니다!